static/src/javascripts/projects/common/modules/article/progress.js

```diff
@@ -0,0 +1,36 @@
+define([
+    'common/utils/$',
+    'common/utils/mediator'
+], function (
+    $,
+    mediator
+) {
+    var articleHeight;
+
+    return {
+        init: function () {
+            // Check if progress bar is present
+            if ($('.progress').length) {
+                this.getArticleHeight();
+                this.bindEvents();
```

Comment:
We're assuming the element is initialized to empty by its style, which seems a bit... couply, maybe? I guess it's okay, but it would be nice to be able to reason about the lifecycle of the element just by looking at one script.